app/src/marketplace/containers/EditProductPage/state.js

```diff
@@ -44,7 +45,7 @@ export const getPendingObject = (product: Product | PendingChanges): Object => {
 }
 
 export const getChangeObject = (original: Product, next: Product): Object => (
-    Object.fromEntries(Object.entries(getPendingObject(next)).filter(([key, value]) => JSON.stringify(value) !== JSON.stringify(original[key])))
+    Object.fromEntries(Object.entries(getPendingObject(next)).filter(([key, value]) => !isEqual(value, original[key])))
```

Comment:
`JSON.stringify` doesn't quarantee deterministic result, the order of the object entries might change which caused it to report changes as pending even though they were not in fact changed.